libgoal/libgoal.go

```diff
@@ -938,6 +940,30 @@ func (c *Client) GetPendingTransactionsByAddress(addr string, maxTxns uint64) (r
 	return
 }
 
+// VerifyParticipationKey checks if a given participationID is installed in a loop until timeout has elapsed.
+func (c *Client) VerifyParticipationKey(timeout time.Duration, participationID string) error {
+	start := time.Now()
+
+	for {
+		keysResp, err := c.GetParticipationKeys()
+		if err != nil {
+			return err
+		}
+		for _, key := range keysResp {
+			if key.Id == participationID {
+				// Installation successful.
+				return nil
+			}
+		}
+
+		if time.Since(start) > timeout {
+			return errors.New("timeout waiting for key to appear")
+		}
+
+		time.Sleep(1 * time.Second)
```

Comment:
1s timeout looks quite high. Is it because `GetParticipationKeys` takes much longer than 1s?

Reply:
I think it is just to be safe.  Some of the tests show that installing some of the participation keys could take longer than 1s so I figured this was a safe value.  The only place this function is called is in the `installParticipationKeyCmd` cobra command in `cmd/goal/account.go` which places the timeout at 1 minute.

If you think it is to high I'm open to suggested values however.

Reply:
I believe this only happens in the context of an admin-only API call that is user initiated. So timing is not very critical.